pjrt/convert/utils/xplane_visitor.zig

```diff
@@ -0,0 +1,293 @@
+const math_utils = @import("math_utils.zig");
+const std = @import("std");
+const xplane_proto = @import("//tsl:xplane_proto");
+const xplane_schema = @import("xplane_schema.zig");
+
+pub const XEventMetadataVisitor = struct {
+    plane: *const XPlaneVisitor,
+    stats_owner: *const xplane_proto.XEventMetadata,
+
+    pub fn init(plane: *const XPlaneVisitor, metadata: *const xplane_proto.XEventMetadata) XEventMetadataVisitor {
+        return .{
+            .plane = plane,
+            .stats_owner = metadata,
+        };
+    }
+
+    pub fn forEachStat(
+        self: *const XEventMetadataVisitor,
+        allocator: std.mem.Allocator,
+        cb: fn (allocator: std.mem.Allocator, xstat: XStatVisitor, ctx: ?*anyopaque) std.mem.Allocator.Error!void,
+        ctx: ?*anyopaque,
+    ) !void {
+        for (self.stats_owner.stats.items) |*stat| {
```

Comment:
I don't really like this visitor pattern. We use it in zml because we need to handle user defined struct, but here you know all the structs, so I'm not sure what you gain compared to writing the for loop every time.

Reply:
Yea, that makes sense; I wrote it this way purely because that's how the XLA implementation handles it, but will rewrite as for loops.